docs/design/simple-rolling-update.md

```diff
@@ -0,0 +1,84 @@
+## Simple rolling update
+This is a lightweight design document for simple rolling update in ```kubectl```
+
+### Lightweight rollout
+Assume that we have a current replication controller named ```foo``` and it is running image ```image:v1```
+
+```kubectl rolling-update rc foo [foo-v2] --image=myimage:v2```
+
+If the user doesn't specify a name for the 'next' controller, then the 'next' controller is renamed to
+the name of the original controller.
+
+Obviously there is a race here, where if you kill the client between delete foo, and creating the new version of 'foo' you might be surprised about what is there, but I think that's ok.
+See Recovery below
+
+If the user does specify a name for the 'next' controller, then the 'next' controller is retained with its existing name,
+and the old 'foo' controller is deleted.
+
+#### Recovery
+If a rollout fails or is terminated in the middle, it is important that the user be able to resume the roll out.
+To facilitate recovery in the case of a crash of the updating process itself, we add the following annotations to each replicaController:
+   * ```desired-replicas``` The desired number of replicas for this controller (either N or zero)
+   * ```update-partner``` A pointer to the resource that is the other half of this update (syntax ```<namespace>:<id>```)
```

Comment:
Is there any realistic case to roll across namespaces right now?  Let's just kick that down the road, I have trouble imagining someone needing it.

> On Apr 21, 2015, at 4:37 PM, Brian Grant <notifications@github.com> wrote:
> 
> In docs/design/simple-rolling-update.md:
> 
> > +```kubectl rolling-update rc foo [foo-v2] --image=myimage:v2```
> > +
> > +If the user doesn't specify a name for the 'next' controller, then the 'next' controller is renamed to
> > +the name of the original controller.
> > +
> > +Obviously there is a race here, where if you kill the client between delete foo, and creating the new version of 'foo' you might be surprised about what is there, but I think that's ok.
> > +See Recovery below
> > +
> > +If the user does specify a name for the 'next' controller, then the 'next' controller is retained with its existing name,
> > +and the old 'foo' controller is deleted.
> > +
> > +#### Recovery
> > +If a rollout fails or is terminated in the middle, it is important that the user be able to resume the roll out.
> > +To facilitate recovery in the case of a crash of the updating process itself, we add the following annotations to each replicaController:
> > +   * ```desired-replicas``` The desired number of replicas for this controller (either N or zero)
> > +   * ```update-partner``` A pointer to the resource that is the other half of this update (syntax ```<namespace>:<id>```)
> I don't think we're using that colon syntax anywhere else. It should be a /. However, I'm not too keen to encourage a pattern where the kind is implicit. Elsewhere I've advocated partial URLs -- at least including the kind.
> 
> That all said, while I don't oppose starting with this approach, I hope it doesn't survive long. We need to move to labels to track the RCs.
> 
> —
> Reply to this email directly or view it on GitHub.
>